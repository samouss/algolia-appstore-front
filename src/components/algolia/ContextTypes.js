import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
