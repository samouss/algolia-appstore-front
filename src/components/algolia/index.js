export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
