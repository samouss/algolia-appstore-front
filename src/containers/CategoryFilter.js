import flowRight from 'lodash.flowright';
import { withFacet, withDisjunctiveFacetRefinement } from 'components/Algolia';
import CategoryList from 'components/CategoryList';

const facet = 'category';

export default flowRight(
  withFacet({ facet }),
  withDisjunctiveFacetRefinement({ facet }),
)(CategoryList);
