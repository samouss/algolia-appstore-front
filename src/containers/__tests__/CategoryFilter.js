import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import CategoryFilter from '../CategoryFilter';

describe('<CategoryFilter />', () => {
  it('expect to render', () => {
    const context = {
      algoliaClient: createMockAlgoliaClient(),
      algoliaHelper: createMockAlgoliaHelper(),
    };

    const component = shallow(
      <CategoryFilter />,
      { context },
    );

    const child = component.shallow({
      context,
    });

    expect(child).toMatchSnapshot();
  });
});
