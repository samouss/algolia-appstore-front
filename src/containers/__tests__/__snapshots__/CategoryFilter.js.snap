// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryFilter /> expect to render 1`] = `
<CategoryList
  facetValues={Array []}
  onChange={[Function]}
/>
`;
